### PR TITLE
Mutex splitter

### PR DESCRIPTION
**Pull request by bouda1**
To simplify tests and use of the splitter object, the read/write mutex is moved from the stream object to the splitter object directly.

### CODE REVIEW

**Comment by bouda1** [comment]
Done.

File: core/test/file/splitter/concurrent.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+  void run() {
+    char* buf = new char[_size];
+    for (int i(0); i < _size; ++i)
+      buf[i] = i & 255;
+
+    for (int j(0); j < _size; j += 100) {
+      QMutexLocker lock(&mutex);
+      int wb(_file->write(buf + j, 100));
+      usleep(rand() % 100);
```
Comment (ganoze) [inline core/test/file/splitter/concurrent.cc]: Shouldn't the mutex be unlocked before usleeping ?

File: core/test/file/splitter/concurrent.cc [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  int             _size;
+
+  void run() {
+    char* buf = new char[_size];
+    for (int i(0); i < _size; ++i)
+      buf[i] = i & 255;
+
+    for (int j(0); j < _size; j += 100) {
+      QMutexLocker lock(&mutex);
+      int wb(_file->write(buf + j, 100));
```
Comment (ganoze) [inline core/test/file/splitter/concurrent.cc]: wb should be used to increment j.

**Comment by bouda1** [comment]
Should be better...

**Pull request merged**